inference/server/main.py

```diff
@@ -106,6 +110,70 @@ def maybe_add_debug_api_keys():
         raise
 
 
+@app.get("/auth/login/discord")
+async def login_discord(request: Request):
+    domain = str(request.url).split("/auth/")[0]
+    redirect_uri = f"{domain}/auth/callback/discord"
+    auth_url = f"https://discord.com/api/oauth2/authorize?client_id={settings.auth_discord_client_id}&redirect_uri={redirect_uri}&response_type=code&scope=identify"
+    raise HTTPException(status_code=302, headers={"location": auth_url})
+
+
+@app.get("/auth/callback/discord", response_model=protocol.Token)
+async def callback_discord(
+    code: str,
+    request: Request,
+    db: sqlmodel.Session = Depends(deps.create_session),
+):
+    domain = str(request.url).split("/auth/")[0]
+    redirect_uri = f"{domain}/auth/callback/discord"
+
+    async with aiohttp.ClientSession(raise_for_status=True) as session:
+        # Exchange the auth code for a Discord access token
+        async with session.post(
+            "https://discord.com/api/oauth2/token",
+            data={
+                "client_id": settings.auth_discord_client_id,
+                "client_secret": settings.auth_discord_client_secret,
+                "grant_type": "authorization_code",
+                "code": code,
+                "redirect_uri": redirect_uri,
+                "scope": "identify",
+            },
+        ) as token_response:
+            token_response_json = await token_response.json()
+            access_token = token_response_json["access_token"]
```

Comment:
raw dict access from outside source is generally dangerous :) maybe some error handling around the whole function?

Reply:
By dangerous do you just mean in terms of the dict maybe not containing the keys we expect? I have implemented quick error handling in that case but let me know if this should be extended further